File: src/main/java/com/team3/gudit/goods/exception/GoodsErrorCodeProvider.java
package com.team3.gudit.goods.exception;

import com.team3.gudit.global.exception.ErrorCode;
import com.team3.gudit.global.exception.ErrorCodeProvider;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class GoodsErrorCodeProvider implements ErrorCodeProvider<GoodsErrorCode> {
    @Override
    public String getDomain() {
        return "GOODS";
    }

    @Override
    public List<ErrorCode> getErrorCodes() {
        return List.of(GoodsErrorCode.values());
    }
}

File: src/main/kotlin/com/team3/gudit/goods/exception/GoodsErrorCodeProvider.kt
package com.team3.gudit.goods.exception

import com.team3.gudit.global.exception.ErrorCode
import com.team3.gudit.global.exception.ErrorCodeProvider
import org.springframework.stereotype.Component

@Component
class GoodsErrorCodeProvider : ErrorCodeProvider<GoodsErrorCode> {
    override fun getDomain(): String = "GOODS"

    override fun getErrorCodes(): List<ErrorCode> = GoodsErrorCode.entries.toList()
}
